tests/parametric/test_otel_span_methods.py

```diff
@@ -286,3 +286,585 @@ def test_otel_get_span_context(self, test_agent, test_library):
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @irrelevant(context.library == "cpp", reason="library does not implement OpenTelemetry")
+    @missing_feature(context.library == "ruby", reason="Not implemented")
+    @missing_feature(context.library == "php", reason="Not implemented")
+    def test_otel_span_operation_name_message_producer(self, test_agent, test_library):
+        """
+            Tests that the operation name will be set to `messaging.system + "." + messaging.operation` when:
+            - Span kind is set to Producer
+            - messaging.system is set to something
+            - messaging.operation is set to something
+
+            (https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/messaging/)
+        """
+        with test_library:
+            with test_library.otel_start_span("otel_span_name", span_kind=SK_PRODUCER) as span:
+                span.set_attributes({"messaging.system": "kafka"})
+                span.set_attributes({"messaging.operation": "publish"})
+                span.end_span()
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, OtelSpan(name="otel_span_name"))
+        assert len(trace) == 1
+
+        root_span = get_span(test_agent)
+
+        assert root_span["name"] == "kafka.publish"
+        assert root_span["resource"] == "otel_span_name"
+
+    @missing_feature(context.library == "go", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @irrelevant(context.library == "cpp", reason="library does not implement OpenTelemetry")
+    @missing_feature(context.library == "ruby", reason="Not implemented")
+    @missing_feature(context.library == "php", reason="Not implemented")
+    def test_otel_span_operation_name_aws_client(self, test_agent, test_library):
+        """
+            Tests that the operation name will be set to aws:
+            - Span kind is set to Client
+            - rpc.system is set to aws-api
+
+            (https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/instrumentation/aws-sdk/)
+        """
+        with test_library:
+            with test_library.otel_start_span("otel_span_name", span_kind=SK_CLIENT) as span:
+                span.set_attributes({"rpc.system": "aws-api"})
+                span.end_span()
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, OtelSpan(name="otel_span_name"))
+        assert len(trace) == 1
+
+        root_span = get_span(test_agent)
+
+        assert root_span["name"] == "aws"  # TODO is there more stuff here that needs to be done?
```

Comment:
It seemed like from the unofficial spec for the operation name that this may need more granularity here on what the operation name should be here, but I'm not sure what

Reply:
My suggestion is to append `otel_span.get_attribute("rpc.service").lower()` to `root_span["name"]` in the tracer: `root_span["name"] = "aws." + otel_span.get_attribute('rpc.service').lower()`:  https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/instrumentation/aws-sdk/#common-attributes

This would yield `aws.s3`, `aws.sns`, for example.

Reply:
Now updated to follow the changes in our doc

`rpc.system`.`rpc.service`.request or `rpc.system`.request (all lowercase)

Ex -> `aws.s3.request` or `aws.request`